### PR TITLE
BDOG-1301: Updates ReadME and refactor Rating Service

### PR DESCRIPTION
**Pull request by Tyrpix**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,19 +4,36 @@
 This service provides metrics about the health of a repository on the Platform.
 
 #### How it works
-The service collects information from various sources which PlatOps deems important to services health, and stores it in a Mongo repository.
+The service collects information from various sources which PlatOps deems important to a service's health, and stores it in a Mongo repository.
```
Comment (cjwebb) [inline README.md:7]: I really dislike the phrase 'which PlatOps deems important'

**Review by cjwebb — DISMISSED** [review]

**Review by cjwebb — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -4,19 +4,37 @@
 This service provides metrics about the health of a repository on the Platform.
 
 #### How it works
-The service collects information from various sources which PlatOps deems important to services health, and stores it in a Mongo repository.
+- Collects information from various sources which are deemed important to a service's health, and stores them in a Mongo repository
```
Comment (anshulbajpai) [inline README.md:7]: Maybe cite a couple of example sources?

File: README.md [hunk 3]
```diff
@@ -4,19 +4,37 @@
...
+- A total score will be calculated that represents the health of a service
 
 # Nomenclature
 - Rater
-  - A Rater is the specific metric a repo is scored on e.g. leak-detection, readME.
-  - Each Rater uses unique functionally depenendent on the metric
-  - These Raters create Ratings
+  - The evaluation of a specific metric for a github repo e.g. leak-detection, readME.
+  - Each Rater uses unique functionally dependent on the metric
+  - Raters create an Indicator
```
Comment (anshulbajpai) [inline README.md:14]: By the sound of it, the Rater and Rating should be closely related but then we have terms like Indicator, Score as well which is making it hard for me to follow the terminology.

Another question to ask ourselves -  **Why Rater creates an Indicator**? Shouldn't it create Rating (if we go by terminology)? Maybe Rater is not a Rater but something else.

File: README.md [hunk 4]
```diff
@@ -4,19 +4,37 @@
...
+  
+- Repository Health Indicator
+   - Each repo has one Repository Health Indicator which is stored in Mongo
```
Comment (anshulbajpai) [inline README.md:17]: An example mongo document will make it clear, perhaps.

File: README.md [hunk 5]
```diff
@@ -4,19 +4,37 @@
...
-  - These can be seen as default Ratings
-- Weight
-  - A weight is a calculation applied to all of a repos Ratings, to produce a *weighted score*
-  - A weighted score is what determines the health of a repo/service
+  - Displays the Service Health of a repo that includes:
+    - Rating Score (total score)
+    - Ratings for each repo:
+        - Rating Type e.g leak-detection, readME
+        - Score given for each rating based on the Indicator
+        - Breakdown of the Score with a description
```
Comment (anshulbajpai) [inline README.md:33]: An example rating will help.

File: app/uk/gov/hmrc/healthindicators/services/RepositoryRatingService.scala [hunk 6]
```diff
@@ -29,25 +29,21 @@ class RepositoryRatingService @Inject() (repository: HealthIndicatorsRepository,
...
-        repositoryRating = RepositoryRating(healthIndicators.repositoryName, repositoryScore, Some(ratings))
-      } yield repositoryRating
+      rate(healthIndicators)
     }
 
-  //TODO: Can you extract a method here?
+  def rate(healthIndicators: Seq[RepositoryHealthIndicator]): Seq[RepositoryRating] =
+    for {
+      healthIndicators <- healthIndicators
+      ratings          = healthIndicators.indicators.map(createRating)
```
Comment (anshulbajpai) [inline app/uk/gov/hmrc/healthindicators/services/RepositoryRatingService.scala:43]: I think it is a single `rating`?

File: README.md (hunk 3 above)
Comment (Tyrpix) [reply README.md:14]: Yeh a Rater creates an Indicator and a Rating is based off the results of that Rater which is inside Mongo(Indicator)

File: app/uk/gov/hmrc/healthindicators/services/RepositoryRatingService.scala [hunk 7]
```diff
@@ -29,25 +29,21 @@ class RepositoryRatingService @Inject() (repository: HealthIndicatorsRepository,
...
-        repositoryScore  = ratings.map(_.ratingScore).sum
-        repositoryRating = RepositoryRating(healthIndicators.repositoryName, repositoryScore, Some(ratings))
-      } yield repositoryRating
+      rate(healthIndicators)
     }
 
-  //TODO: Can you extract a method here?
+  def rate(healthIndicators: Seq[RepositoryHealthIndicator]): Seq[RepositoryRating] =
+    for {
+      healthIndicators <- healthIndicators
```
Comment (anshulbajpai) [inline app/uk/gov/hmrc/healthindicators/services/RepositoryRatingService.scala:42]: Should it be `healthIndicator <- healthIndicators` ?

File: README.md (hunk 3 above)
Comment (anshulbajpai) [reply README.md]: This is something with which I am still not convinced. The output of Rater is Indicator here. I would expect Rater to produce Rate or Rating or something which sounds similar to Rater. There is something here that I think is not correctly named. But naming things is not easy. I'll let you think more on it and maybe with future iterations of this code, we might have a better name.

**Review by anshulbajpai — APPROVED** [review]

**Pull request merged**